File: year2022/day06/java/Solution.java
import java.nio.file.*;
import java.util.*;

public class Solution {
    private static String[] CHARS;

    public static void main(String[] args) { 
        try {
            setup(Files.readString(Paths.get("../puzzle.txt")));
            System.out.println(part1());
            System.out.println(part2());
        } catch(Exception e) {
            e.printStackTrace();
        }
    }
    
	public static Integer part1() {
        return getMarker(4);
    }
    
    public static Integer part2() {
        return getMarker(14);
    }

    public static void setup(String input) {
        CHARS = input.split("");
    }

    private static int getMarker(int len) {
        for (int i = 0; i < CHARS.length - 3; i++) {
            List<String> key = new ArrayList<String>();
            for (int j = 0; j <= len - 1; j++) {
                key.add(CHARS[i + j]);
            }
            if (new HashSet<>(key).size() == key.size()) {
                return i + len;
            }
        }
        return -1;
    }
}

File: year2022/day06/python/solution.py
def main():
  print(part_1())
  print(part_2())


def part_1():
  return getMarker(input, 4)


def part_2():
  return getMarker(input, 14)


def setup(path):
  global input

  with open(path, 'r') as file:
    input = ([*file.read()])


def getMarker(input, length):
  return next((i + length for i, _ in enumerate(input[:-length]) if len(set(input[i:i+length])) == length), None)


if __name__ == '__main__':
  setup('../puzzle.txt')
  main()